packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/snapshot_list.dart

```diff
@@ -132,6 +100,43 @@ class _SnapshotListTitle extends StatelessWidget {
   }
 }
 
+class _SnapshotListItems extends StatelessWidget {
+  _SnapshotListItems({Key? key, required this.controller}) : super(key: key);
+
+  final DiffPaneController controller;
+  final headerHeight = 1.20 * defaultRowHeight;
+
+  @override
+  Widget build(BuildContext context) {
+    return ValueListenableBuilder<List<DiffListItem>>(
+      valueListenable: controller.snapshots,
+      builder: (_, list, __) {
+        return ListView.builder(
+          controller: controller.scrollController,
+          shrinkWrap: true,
+          itemCount: controller.snapshots.value.length,
+          itemBuilder: (context, index) {
+            return Container(
+              height: headerHeight,
+              color: controller.selectedIndex.value == index
```

Comment:
I'm not seeing where this widget is rebuilding when `controller.selectedIndex.value` changes now? _SnapshotListItems needs to rebuild when the snapshot list changes and when the index changes. Let's make this widget stateful, and add listeners for both `controller.snapshots` and `controller.selectedIndex`, calling setState when either change and also scrolling to the bottom of the list when the snapshots list changes